### ISSUE TITLE
How to make Norch and Search-index work at the same time

### ISSUE THREAD

**Issue by eklem** [fergiemcdowall/search-index#79]
I'm running `norch` with `norch-bootstrap` from a console and I'm trying to use `search-index`from within a JavaScript to add documents to the index while running `norch`. I'm getting a conflict:

```
events.js:85
      throw er; // Unhandled 'error' event
            ^
OpenError: IO error: lock si/LOCK: Resource temporarily unavailable
    at /home/pi/node_modules/norch/node_modules/search-index/node_modules/levelup/lib/levelup.js:118:34
```

If I stop `norch`, the script works, and documents are added to the index.

So my question: How to run `norch` and use the `search-index` API at the same time?

**Comment by eklem** [comment]
So, a bit more on my use case. Really simple: I want to run Norch as in
```Console
pi@fatboy ~ $ node node_modules/.bin/norch
```
... and at the same time be able to index new documents. I'm using `si.add` from `search-index` quite straight forward. Gather docs, document process them and run the whole batch in one go:
```JavaScript
// Modules and stuff required
var gsheets = require('gsheets')
var fs = require('fs-extra')
var ifttnorch = require('iftt-norch-tools')
var si = require('search-index')

// Read config file
var data = fs.readFileSync('/home/pi/node_modules/life-indexer/config/config-twitter.json'), config
config = JSON.parse(data)

// Get csv-file as 'data' (object)
gsheets.getWorksheet(config.gsheetsKey, config.gsheetsWorksheet, function(err, result) {
  console.log(result.data)
  // Iterating through rows, adding and adjusting
  for (var i=0; i<result.data.length; i++) {
    var obj = result.data[i]

    // Date extraction and transformation
    obj.date = ifttnorch.date(obj.date)
    obj.id = ifttnorch.id(obj.date, obj.text, obj.type)
    obj.user = ifttnorch.twitterusers(obj.user, obj.text)
    obj.tags = ifttnorch.tags(obj.text)
    obj.links = ifttnorch.links(obj.text)
    obj.type = [config.type]
    //obj = [obj]

    //var resultJSON = resultJSON.escapeUnescape()

    console.log('Object #' + i)
    console.log(obj + '\n\n\n')
  }
  console.log(result.data)

  si.add({'batchName': config.batchname, 'filters': config.filters}, result.data, function(err) {
    if (!err) console.log('indexed!')
  })
});

// Function for escaping and unescaping characers from JSON.stringify-process
// http://stackoverflow.com/questions/4253367/how-to-escape-a-json-string-containing-newline-characters-using-javascript
String.prototype.escapeUnescape = function() {
  return this.replace(/\\n/g, "\\n")
             .replace(/\\'/g, "\\'")
             .replace(/\\"/g, '\\"')
             .replace(/\\&/g, "\\&")
             .replace(/\\r/g, "\\r")
             .replace(/\\t/g, "\\t")
             .replace(/\\b/g, "\\b")
             .replace(/\\f/g, "\\f");
};
```

**Comment by eklem** [comment]
In plain english: Run Norch and index documents at the same time =)

**Comment by mewwts** [comment]
Thanks @eklem, and sorry to disappoint you, but I can't get this done before the weekend :-1:

**Comment by eklem** [comment]
No problem. I'll have plenty of other stuff I can do to make the `Life-Index` project move forward :+1:

**Comment by eklem** [comment]
A little question that may help. How does the curl indexing method work in Norch?
https://github.com/fergiemcdowall/norch#http-interface

**Comment by fergiemcdowall** [comment]
One of the limitations of search-index (and any log structured merge tree) is that the "owner" process needs to know everything about it. Trying to access a levelup instance that is already in use results in woe.

Is there an underlying issue here or are you doing this in the name of mad science?

**Comment by eklem** [comment]
Not mad science, I think. I want to run Norch and Norch-bootstrap since it helps me with a lot of stuff I don't want to care about or build myself. But for the indexing part, I want to use search-index si.add since that gives me the level of control I need.

I guess I want to use the newbie-stuff for running a web server and frontend, and the advanced stuff for indexing, and that doesn't go together too well.

Also indexing through http-interface with curl from inside a script just seemed too technical for me.

**Comment by mewwts** [comment]
This should be fixed now! :dancers:

**Comment by eklem** [comment]
This is just so good =)
